presentation-api/controlling-ua/reconnectToPresentation_success-manual.html

```diff
@@ -1,72 +1,49 @@
-    // Helper Function - start the presentation
-    // ----------------------------------------
-    var startPresentation = function () {
-        return request.start()
-                .then(function (connection) {
-                    // save the presentation id within the
-                    // local storage of the browser
-                    localStorage["presId"] = connection.id;
+    async_test(function (t) {
+        var startBtn = document.getElementById("startBtn");
+        startBtn.onclick = t.step_func(function () {
+            startBtn.disabled = true;
+            var request = new PresentationRequest(presentationUrls);
+            // start presentation
+            request.start().then(t.step_func(function (oldConnection) {
+                // presentationId will be used to reconnect to presentation
+                var presentationId = oldConnection.id;
+                var oldConnectionClosed = false;
+                oldConnection.onclose = t.step_func(function () {
+                    oldConnectionClosed = true;
+                    // reconnect to presentation using presentationId
+                    request.reconnect(presentationId).then(t.step_func(function (newConnection) {
+                        assert_equals(newConnection.state, 'connecting',"connection should be in 'connecting' state");
+                        assert_equals(newConnection.id, presentationId, "Ids of old and new connections must be equal");
+                        t.done();
+                    })).catch(t.step_func(function () {
+                        assert_unreached("Reconnect to presentation promise rejected");
+                    }));
                 });
-    }
-
-    // ------------------------------------------------
-    // Reconnect to Presentation Test (success) - begin
-    // ------------------------------------------------
-    var reconnectToPresentation = function () {
-
-        promise_test(function () {
-
-            // get the saved presentation id
-            // for the reconnect function if exist
-            if (localStorage["presId"]) var presId = localStorage["presId"];
-
-            return request.reconnect(presId)
-                    .then(function (connection) {
-                        assert_true(connection instanceof PresentationConnection);
-                        done();
-                    });
-        }, 'The reconnection was successful.');
-    }
-    // ----------------------------------------------
-    // Reconnect to Presentation Test (success) - end
-    // ----------------------------------------------
+                // close connection. test continue after onclose event is fired.
+                oldConnection.close();
+                // the test fails if after 5s the close event is not fired
+                t.step_timeout(function () {
+                    !oldConnectionClosed && assert_unreached("Close event on old connection not fired");
```

Comment:
The assert will make the test `FAIL` but I would rather expect the test to fail with a `TIMEOUT` status.

The "Harness timeout" section in the [Testharness.js API Documentation](http://testthewebforward.org/docs/testharness-library.html) mentions the `force_timeout` method (not exactly for the same purpose and as "last resort"): "it is possible to use `test.force_timeout()` in place of `assert_unreached()`, to immediately fail the test but with a status of `TIMEOUT`".

Beware though, you need to call `t.done()` as well after the call to `t.force_timeout()`.